rpcserver.go

```diff
@@ -4749,3 +4756,310 @@ func MarshalAssetFedSyncCfg(
 		AllowSyncExport: config.AllowSyncExport,
 	}, nil
 }
+
+// unmarshalAssetSpecifier unmarshals an asset specifier from the RPC form.
+func unmarshalAssetSpecifier(req *rfqrpc.AssetSpecifier) (*asset.ID,
+	*btcec.PublicKey, error) {
+
+	// Attempt to decode the asset specifier from the RPC request. In cases
+	// where both the asset ID and asset group key are provided, we will
+	// give precedence to the asset ID due to its higher level of
+	// specificity.
+	var (
+		assetID *asset.ID
+
+		groupKeyBytes []byte
+		groupKey      *btcec.PublicKey
+
+		err error
+	)
+
+	switch {
+	// Parse the asset ID if it's set.
+	case len(req.GetAssetId()) > 0:
+		var assetIdBytes [32]byte
+		copy(assetIdBytes[:], req.GetAssetId())
+		id := asset.ID(assetIdBytes)
+		assetID = &id
+
+	case len(req.GetAssetIdStr()) > 0:
+		assetIDBytes, err := hex.DecodeString(req.GetAssetIdStr())
+		if err != nil {
+			return nil, nil, fmt.Errorf("error decoding asset "+
+				"ID: %w", err)
+		}
+
+		copy(assetID[:], assetIDBytes)
```

Comment:
nit: nil deref warning, something like this should work:

```go
var id asset.ID
copy(id[:], assetIDBytes)
assetID = &id
```

Reply:
Thanks for the code.